### PR TITLE
Add Pattern Prop to Input Component

### PR DESCRIPTION
**Pull request by amichaelparker**
### Summary
This adds the html pattern attribute as a first class prop to terra-form-input. 

Closes [#2089](https://github.com/cerner/terra-core/issues/2089)

### CODE REVIEW

File: packages/terra-form-input/src/Input.jsx [hunk 1]
```diff
@@ -40,6 +40,10 @@ const propTypes = {
    * Content to be displayed as the name.
    */
   name: PropTypes.string,
+  /**
+   * Pattern to match when input type is used.
```
Comment (JakeLaCombe) [inline packages/terra-form-input/src/Input.jsx]: This comment is a little confusing with the is used ending. I would just say Pattern for the input value.

**Comment by JakeLaCombe** [comment]
I would expect a jest test to be added and updated

File: packages/terra-form-input/src/Input.jsx (hunk 1 above)
Comment (amichaelparker) [reply packages/terra-form-input/src/Input.jsx]: Comment changed to suggested text.

File: packages/terra-form-input/tests/jest/Input.test.jsx [hunk 2]
```diff
@@ -16,6 +16,12 @@ describe('Input', () => {
     expect(wrapper).toMatchSnapshot();
   });
 
+  it('should render with a type of password and pattern prop value', () => {
+    const input = <Input type="password" pattern=".{6,}" />;
+    const wrapper = render(input);
```
Comment (StephenEsser) [inline packages/terra-form-input/tests/jest/Input.test.jsx]: Is there a specific reason to do a full render vs a shallow render here?

Comment (amichaelparker) [reply packages/terra-form-input/tests/jest/Input.test.jsx]: I updated that test (and the others) to run shallow, and the snaps all seem to look good. Look them over and let me know!

File: packages/terra-form-input/src/Input.jsx [hunk 3]
```diff
@@ -40,6 +40,10 @@ const propTypes = {
    * Content to be displayed as the name.
    */
   name: PropTypes.string,
+  /**
+   * Pattern for the input value.
```
[the remark this answers is not in the document]
Comment (emilyrohrbough) [reply packages/terra-form-input/src/Input.jsx]: According to the definition, this pattern applies to certain input types. Should we also add the type prop so it is also first class?

[the remark this answers is not in the document]
Comment (bjankord) [reply packages/terra-form-input/src/Input.jsx]: Sure

[the remark this answers is not in the document]
Comment (bjankord) [reply packages/terra-form-input/src/Input.jsx]: Let's get an issue logged for adding type as a first-class prop and handle it in a new PR.

**Pull request merged**